crates/ra_mbe/src/syntax_bridge.rs

```diff
@@ -12,12 +12,34 @@ use tt::buffer::{Cursor, TokenBuffer};
 
 use crate::subtree_source::SubtreeTokenSource;
 use crate::ExpandError;
+use std::sync::atomic::{AtomicU32, Ordering};
 
 /// Maps `tt::TokenId` to the relative range of the original token.
-#[derive(Default)]
+#[derive(Debug, PartialEq, Eq)]
 pub struct TokenMap {
     /// Maps `tt::TokenId` to the *relative* source range.
     tokens: Vec<TextRange>,
+    map_id: u32,
+}
+
+static TOKEN_MAP_COUNTER: AtomicU32 = AtomicU32::new(0);
+
+/// Generate an unique token map id for each instance
+fn make_uniq_token_map_id() -> u32 {
+    TOKEN_MAP_COUNTER.fetch_add(1, Ordering::SeqCst)
+}
+
+impl std::default::Default for TokenMap {
+    fn default() -> TokenMap {
+        TokenMap { tokens: Default::default(), map_id: make_uniq_token_map_id() }
+    }
+}
+
+/// Maps Relative range of the expanded syntax node to `tt::TokenId`
+#[derive(Debug, PartialEq, Eq, Default)]
+pub struct ExpandedRangeMap {
+    /// Maps `tt::TokenId` to the *relative* source range.
+    ranges: Vec<(TextRange, tt::TokenId)>,
```

Comment:
The order of the tuple elems seems to imply that TextRange gets mapped to TokenId while the doc says otherwise.

Reply:
Fixed, Thanks